### ISSUE TITLE
Terminal corruption

### ISSUE THREAD

**Issue by henryiii** [pyodide/pyodide#2392]
## 🐛 Bug

The terminal on can get corrupted with the following code:

```python
import micropip
await micropip.install("click")
click.get_text_stream("stdout")
```

You can see the result (in place, before I worked out where it was coming from):

<img width="788" alt="Screen Shot 2022-04-11 at 5 17 18 PM" src="https://user-images.githubusercontent.com/4616906/162835072-cd488cd8-866d-4eac-ba45-f2fa461a48d8.png">

Notice the `b''` printout. After that, the terminal doesn't behave correctly. The final newline doesn't work and there are some other quirks. You can see a bit of it in the next line above.

I should be able to fix this in my case by proposing a fix for histoprint; it does not need to be importing click and messing with this if used as a library, I think. CC @ast0815.

### To Reproduce

See above, can be run in https://pyodide.org/en/stable/console.html

### Expected behavior

No `b""`, and no corruption.

### Environment

- Pyodide Version: 0.20.0
- Browser version: Originally Safari, but reproduced on Chrome.
- Any other relevant information:

**Comment by henryiii** [comment]
This sounds relevant: https://click.palletsprojects.com/en/7.x/python3/#python-3-limitations

**Comment by rth** [comment]
Yes, I can reproduce, it does indeed partially breaks the REPL. Though I imagine anything that changes the stdout stream would be able to mess with the REPL.  click seems to be doing a lot of complicated things with the stdout/stderr streams
https://github.com/pallets/click/blob/bf9da4838986af3bc09a1b16974d154c61dcbe3f/src/click/_compat.py#L243

**Comment by henryiii** [comment]
Not quite related, but close: I guess `termios` is known to be missing?

**Comment by hoodmane** [comment]
Yeah. It is possible to build `termios` but I think it might be completely useless.

**Comment by henryiii** [comment]
Yeah, what I was trying there is likely very much not going to work, but I noticed it was missing. Not related to click muddling up the output at all.
